.github/workflows/create-pr.yml

```diff
@@ -0,0 +1,119 @@
+# This workflow is triggered when the remote dependency repos request it.
+# It will create a PR with the new dependency and the new branch.
+# The PR will be created from the develop branch.
+# Needed env variables:
+#   PULL_REQUEST_GIT_USER - The git user that will be used to create the branch and the PR.
+#   ACCEPTED_PACKAGES - A JSON array of accepted package names.
+name: Create PR with external dependency
+
+on:
+  repository_dispatch:
+    types: [create_pr]
+
+jobs:
+  build:
+    name: Create PR with external dependency
+    runs-on: ubuntu-latest
+    steps:
+      - name: Verify Event and set env
+        env:
+          PR_BRANCH: ${{ github.event.client_payload.branch }}
+          PACKAGE: ${{ github.event.client_payload.dependency_package }}
+        run: |
+          PACKAGE_NAME=$(echo "${{ github.event.client_payload.dependency_package }}" | sed 's/[^\/]*\///')
+          echo "PR_BRANCH=$PR_BRANCH" >> $GITHUB_ENV
+          echo "PACKAGE=$PACKAGE" >> $GITHUB_ENV
+          echo "PACKAGE_NAME=$PACKAGE_NAME" >> $GITHUB_ENV
+
+          PACKAGE_NAME_IS_VALID=$(echo '${{ vars.ACCEPTED_PACKAGES }}' | jq --arg project "${{ env.PACKAGE }}" 'any(.[]; . as $name | $project | match($name))')
+          echo "PACKAGE_NAME_IS_VALID=$PACKAGE_NAME_IS_VALID" >> $GITHUB_ENV
+        if: ${{ github.event.action == 'create_pr' && github.event.client_payload.branch && github.event.client_payload.dependency_package }}
+
+      - name: Stop workflow if any variables are missing
+        run: |
+          echo "::error::Some variable is missing. Have to stop..."
+          exit 1
+        if: ${{ !env.PR_BRANCH || !env.PACKAGE || !env.PACKAGE_NAME}}
+
+      - name: Stop workflow if package name is not valid
+        run: |
+          echo "::error::The package name is not valid. Should match one of the accepted package names."
+          exit 1
+        if: ${{ env.PACKAGE_NAME_IS_VALID != 'true' }}
+
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - id: check_branch_exists
+        uses: GuillaumeFalourd/branch-exists@v1
+        with:
+          branch: ${{ env.PR_BRANCH }}
+
+      - name: Stop workflow if branch already exists
+        run: |
+          echo "::error::A branch with the same name as the requested branch name already exists."
+          exit 1
+        if: ${{ steps.check_branch_exists.outputs.exists == 'true' }}
```

Comment:
I wonder if *failing* the workflow is necessary it does not seem as if any error has actually occurred. On the other hand just stopping the workflow might also be tricky....

Reply:
Don't know exactly what you mean. But `exit 1` makes sure to stop the workflow in case a branch already exists.